### PR TITLE
Social auth

### PR DESCRIPTION
**Pull request by adelajaOlamilekan**
​
## Description
This PR comes with the Google social authentication feature

​
## Related Issue (Link to linear ticket)

​
## Motivation and Context
This PR is needed to ensure that the application provides SSO(Single Sign-on) feature.

​
## How Has This Been Tested?

​
## Screenshots (if appropriate - Postman, etc):
​
Screenshot (1295)
Screenshot (1297)
Screenshot (1296)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/routes/auth_routes.py [hunk 1]
```diff
@@ -137,3 +174,109 @@ async def logout_user(
...
+    if user_is_loggedin:
+        return UserResponse(
+            status_code=401, message="A User is Already Logged in", data=None
+        )
+    
+    with google_sso:
+        user = await google_sso.verify_and_process(request)
+
+    if not user:
+        UserResponse(status_code=400, message="Failed to Login from Google", data=None)
```
Comment (zxenonx) [inline app/routes/auth_routes.py:210]: return statement missing here

File: app/routes/auth_routes.py [hunk 2]
```diff
@@ -137,3 +174,109 @@ async def logout_user(
...
+    if user_logged_in:
+             return UserResponse(
+            status_code=401, message="User Already Logged in", data=None
+        )
+
+    with facebook_sso:
+        user = await facebook_sso.verify_and_process(request)
+    
+    if not user:
+        UserResponse(status_code=400, message="Failed to Login from Google", data=None)
```
Comment (zxenonx) [inline app/routes/auth_routes.py:263]: return statement missing

File: app/settings.py [hunk 3]
```diff
@@ -4,7 +4,8 @@
 config = ConfigParser()
 config.read("config.ini")
 
-DEEPGRAM_API_KEY = config["deepgram"]["api_key"]
+DEEPGRAM_API_KEY= ""
+#config["deepgram"]["api_key"]
```
Comment (zxenonx) [inline app/settings.py:8]: Can we remove this?

File: app/routes/auth_routes.py [hunk 4]
```diff
@@ -137,3 +174,109 @@ async def logout_user(
...
+    request.session["logged_in"] = True
+
+    return UserResponse (status_code=200, message="User Logged in Successfuly", data=None)
+
+@auth_router.get("/facebook/login")
+async def facebook_login(request: Request):
+    """Generate Login for the User"""
+
+    with facebook_sso:
+        return  await facebook_sso.get_login_redirect()
```
Comment (zxenonx) [inline app/routes/auth_routes.py:237]: remove the extra space between "return" and "await"

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @adelajaOlamilekan

File: app/routes/auth_routes.py (hunk 1 above)
Comment (adelajaOlamilekan) [reply app/routes/auth_routes.py:210]: Ok on it

File: app/settings.py (hunk 3 above)
Comment (adelajaOlamilekan) [reply app/settings.py:8]: Sure

**Pull request merged**